core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -45,6 +45,33 @@
 
 public class TableMetadataParser {
 
+  public enum Codec {
+    NONE(""),
+    GZIP(".gz");
+
+    private final String extension;
+
+    Codec(String extension) {
+      this.extension = extension;
+    }
+
+    public static Codec fromName(String codecName) {
+      Preconditions.checkArgument(codecName != null, "Codec name is null");
+      return Codec.valueOf(codecName.toUpperCase(Locale.ENGLISH));
+    }
+
+    public static Codec fromFileName(String fileName) {
```

Comment:
Should this be backward-compatible with the existing `.metadata.json.gz` file names?

Reply:
Done